### PR TITLE
JE-69: Users

### PR DESCRIPTION
**Pull request by rimi-itk**
The site is split into three parts:

1. “portal” (path: `/`) – the public area (after login, though)
2. “jira“ (path: `jira/`) – protected by OAuth login via Jira. 
3. “admin” (path: `admin`) – admin area

* Regular user login (via fos user-bundle) required to access site
* Users with the role ROLE_JIRA can access /jira where OAuth login via Jira is required
* MenuService reads menu items from configuration (actually, parameters defined in `jira_economics.yaml) and checks access based on user roles
* An AppService has been added to provide apps – this service is very similar to MenuService.
* An admin area (based on EasyAdmin) has been added. For now, it's just used to manage users.

Known issues

* Creating users via the admin ui doesn't yet work

### CODE REVIEW

File: src/Twig/AppExtension.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+namespace App\Twig;
+
+use App\Service\ContextService;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+class AppExtension extends AbstractExtension
+{
+    private const PATH_JIRA = 'jira';
+    private const PATH_PORTAL = 'jira';
```
Comment (tuj) [inline src/Twig/AppExtension.php]: Why both 'jira'?

**Comment by tuj** [comment]
After login I get the following error when accessing /create-graphic-service-order/new
Screenshot 2019-08-08 at 09 56 12

**Review by tuj — CHANGES_REQUESTED** [review]
Generally the code looks good. I have a couple of comments.
* The login page is not "styled".
* There are some untranslated strings here and there.

File: src/Twig/AppExtension.php (hunk 1 above)
Comment (rimi-itk) [reply src/Twig/AppExtension.php]: Because I'm a bad programmer. And these Twig extensions are not needed at all. I'll remove them.

**Comment by rimi-itk** [comment]
@tuj , please re-re-re-review …

File: templates/base.html.twig [hunk 2]
```diff
@@ -36,17 +36,19 @@
...
+                {% include 'navigation/global-nav.html.twig' %}
             </div>
         {% endblock %}
         {% block main %}
             <div class="d-flex flex-row flex-grow-1">
                 <div class="container-fluid">
                     <div class="main row">
-                        {# Only render sidebar (and wrapper) if it is defined #}
-                        {% if block('sidebar') is defined %}
+                        {% block sidebar '' %}
```
Comment (tuj) [inline templates/base.html.twig]: This file should be the develop version instead.

File: templates/navigation/global-nav.html.twig [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
-                            <a class="nav-link{{ globalMenuItem.active ? ' active' : '' }} pl-0 pr-0" href="{{ path(globalMenuItem.routeName) }}" data-toggle="tooltip" data-placement="right" title="{{ globalMenuItem.desc }}">
-                                <i class="fas {{ globalMenuItem.icon }} fa-fw" ></i><span class="d-md-block ml-3 ml-md-0">{{ globalMenuItem.title }}</span>
-                            </a>
-                        </li>
-                    {% endfor %}
-                {% endif %}
-            </ul>
-        </div>
-    </nav>
+  <nav class="navbar navbar-expand-md navbar-dark bg-primary flex-md-column p-md-1">
```
Comment (tuj) [inline templates/navigation/global-nav.html.twig]: 4 indentions

File: templates/portal/base.html.twig [hunk 4]
```diff
@@ -0,0 +1,89 @@
+<!DOCTYPE html>
```
Comment (tuj) [inline templates/portal/base.html.twig]: Is this the same file as base.html.twig?

**Review by tuj — CHANGES_REQUESTED** [review]

File: templates/portal/base.html.twig (hunk 4 above)
Comment (rimi-itk) [reply templates/portal/base.html.twig]: This file is not needed. I'll delete it.

**Review by tuj — APPROVED** [review]

**Pull request merged**